src/taskgraph/run-task/run-task

```diff
@@ -1061,6 +1069,9 @@ def main(args):
     parser.add_argument("--user", default="worker", help="user to run as")
     parser.add_argument("--group", default="worker", help="group to run as")
     parser.add_argument("--task-cwd", help="directory to run the provided command in")
+    parser.add_argument("--shallow", type=lambda x: x.lower() == "true",
```

Comment:
Instead of using a lambda with `type`, you should specify `action="store_true"`. This will allow you to pass `--shallow` without any argument at all.